src/main/resources/handlebars/JavaMicronaut/beanValidationParamsInner.mustache

```diff
@@ -0,0 +1,5 @@
+{{^isPrimitiveType}}{{^isEnum}}{{!
+    Non-container
+}}{{^isContainer}}@Valid {{/isContainer}}{{!
+    Container
+}}{{#isContainer}}{{#items}}{{>beanValidationParamsInner}}{{/items}}{{/isContainer}}{{/isEnum}}{{/isPrimitiveType}}
```

Comment:
Had to introduce recursive template :( But I do not see any other way to implement this. Looks like Micronaut throws `ConstraintViolationException` saying something about absent introspection if `List` of primitive types is marked as `@Valid`. This is there to handle `List<List<String>>` case.